src/tests/tracing/eventcounter/eventcounter.csproj

```diff
@@ -4,7 +4,7 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <GCStressIncompatible>true</GCStressIncompatible>
     <!-- This test is timing sensitive and JIT timing affects the results of the test -->
-    <JitOptimizationSensitive>true</JitOptimizationSensitive>
+    <JitOptimizationSensitive Condition="'$(TargetOS)' != 'Android'">true</JitOptimizationSensitive>
```

Comment:
Is this just to get the test running on the Android CoreCLR runtime test configuration? Believe this should be set when we run the runtime tests used the same configuration as we do on OSX, it should set the COM_TieredCompilation=0 as part of running no_tiered_compilation, that in turn should be part of the innerloop configuration, https://github.com/dotnet/runtime/blob/4c662b16e3e6fe688f9bbff2a9423fdee799ecfe/eng/pipelines/runtime.yml#L1062. Similar to what we did in this PR, https://github.com/dotnet/runtime/pull/58106

Having that said it might be equal important to be less dependent on all these configs and since this is not an issue on MonoVM (not running tiered JIT), we could just make sure we run the tests independently.

Reply:
Good catch. Now that I'm inspecting this, I do believe I had been running the test on Android CoreCLR runtime test configuration. I thought it had been necessary to exclude `JitOptimizationSensitive` to run the test, but I guess its a CoreCLR specific property.

I thought I was running on mono runtime with 
`<reporoot>/build.sh -s mono+libs -os Android -arch x64 -c Debug`
`<reporoot>/src/tests/build.sh /p:LibrariesConfiguration=Debug -excludemonofailures -rebuild mono -os Android debug -test:tracing/eventcounter/eventcounter.csproj`
`<reporoot>/dotnet.sh artifacts/tests/coreclr/Android.x64.Debug/Tests/Core_Root/xunit.console.dll artifacts/tests/coreclr/Android.x64.Debug/tracing/eventcounter/tracing.eventcounter.XUnitWrapper.dll`
But this is hitting `SKIPPING EXECUTION BECAUSE COMPlus_TieredCompilation has not been disabled and this test is marked JitOptimizationSensitive
Test Harness Exitcode is : 0` so it must still be CoreCLR?

I did see `RuntimeFlavor=mono` in some of the console text when building the test, what am I missing to run it as mono? (I'm assuming its not mono if i'm still seeing the output above'
```
Building managed test group 1: "/Users/mdhwang/runtime_droid/dotnet.sh" msbuild /Users/mdhwang/runtime_droid/src/tests/build.proj /t:Build "/p:TargetArchitecture=x64" "/p:Configuration=Debug" "/p:LibrariesConfiguration=Debug" "/p:TargetOS=Android" "/p:RuntimeOS=" "/p:RuntimeFlavor=mono" "/p:RuntimeVariant=" "/p:CLRTestBuildAllTargets=" "/p:__TestGroupToBuild=1" "/p:__SkipRestorePackages=1" /nodeReuse:false /maxcpucount /bl:/Users/mdhwang/runtime_droid/artifacts//log/Debug/InnerManagedTestBuild.1.binlog
  Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET
  Copyright (C) Microsoft Corporation. All rights reserved.

  /Users/mdhwang/runtime_droid/.dotnet/sdk/6.0.100/MSBuild.dll -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,/Users/mdhwang/runtime_droid/.dotnet/sdk/6.0.100/dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,/Users/mdhwang/runtime_droid/.dotnet/sdk/6.0.100/dotnet.dll -maxcpucount /maxcpucount -verbosity:m /bl:/Users/mdhwang/runtime_droid/artifacts//log/Debug/InnerManagedTestBuild.1.binlog /nodeReuse:false /p:TargetArchitecture=x64 /p:Configuration=Debug /p:LibrariesConfiguration=Debug /p:TargetOS=Android /p:RuntimeOS= /p:RuntimeFlavor=mono /p:RuntimeVariant= /p:CLRTestBuildAllTargets= /p:__TestGroupToBuild=1 /p:__SkipRestorePackages=1 /t:Build /Users/mdhwang/runtime_droid/src/tests/build.proj
  Building managed test group 2: "/Users/mdhwang/runtime_droid/dotnet.sh" msbuild /Users/mdhwang/runtime_droid/src/tests/build.proj /t:Build "/p:TargetArchitecture=x64" "/p:Configuration=Debug" "/p:LibrariesConfiguration=Debug" "/p:TargetOS=Android" "/p:RuntimeOS=" "/p:RuntimeFlavor=mono" "/p:RuntimeVariant=" "/p:CLRTestBuildAllTargets=" "/p:__TestGroupToBuild=2" "/p:__SkipRestorePackages=1" /nodeReuse:false /maxcpucount /bl:/Users/mdhwang/runtime_droid/artifacts//log/Debug/InnerManagedTestBuild.2.binlog
  Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET
  Copyright (C) Microsoft Corporation. All rights reserved.

  /Users/mdhwang/runtime_droid/.dotnet/sdk/6.0.100/MSBuild.dll -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,/Users/mdhwang/runtime_droid/.dotnet/sdk/6.0.100/dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,/Users/mdhwang/runtime_droid/.dotnet/sdk/6.0.100/dotnet.dll -maxcpucount /maxcpucount -verbosity:m /bl:/Users/mdhwang/runtime_droid/artifacts//log/Debug/InnerManagedTestBuild.2.binlog /nodeReuse:false /p:TargetArchitecture=x64 /p:Configuration=Debug /p:LibrariesConfiguration=Debug /p:TargetOS=Android /p:RuntimeOS= /p:RuntimeFlavor=mono /p:RuntimeVariant= /p:CLRTestBuildAllTargets= /p:__TestGroupToBuild=2 /p:__SkipRestorePackages=1 /t:Build /Users/mdhwang/runtime_droid/src/tests/build.proj
  Building managed test group 3: "/Users/mdhwang/runtime_droid/dotnet.sh" msbuild /Users/mdhwang/runtime_droid/src/tests/build.proj /t:Build "/p:TargetArchitecture=x64" "/p:Configuration=Debug" "/p:LibrariesConfiguration=Debug" "/p:TargetOS=Android" "/p:RuntimeOS=" "/p:RuntimeFlavor=mono" "/p:RuntimeVariant=" "/p:CLRTestBuildAllTargets=" "/p:__TestGroupToBuild=3" "/p:__SkipRestorePackages=1" /nodeReuse:false /maxcpucount /bl:/Users/mdhwang/runtime_droid/artifacts//log/Debug/InnerManagedTestBuild.3.binlog
  Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET
  Copyright (C) Microsoft Corporation. All rights reserved.

  /Users/mdhwang/runtime_droid/.dotnet/sdk/6.0.100/MSBuild.dll -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,/Users/mdhwang/runtime_droid/.dotnet/sdk/6.0.100/dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,/Users/mdhwang/runtime_droid/.dotnet/sdk/6.0.100/dotnet.dll -maxcpucount /maxcpucount -verbosity:m /bl:/Users/mdhwang/runtime_droid/artifacts//log/Debug/InnerManagedTestBuild.3.binlog /nodeReuse:false /p:TargetArchitecture=x64 /p:Configuration=Debug /p:LibrariesConfiguration=Debug /p:TargetOS=Android /p:RuntimeOS= /p:RuntimeFlavor=mono /p:RuntimeVariant= /p:CLRTestBuildAllTargets= /p:__TestGroupToBuild=3 /p:__SkipRestorePackages=1 /t:Build /Users/mdhwang/runtime_droid/src/tests/build.proj
  ```

Reply:
I'll remove all the associated `JitOptimizationSensitive` exclusions and check CI results again.

> Having that said it might be equal important to be less dependent on all these configs and since this is not an issue on MonoVM (not running tiered JIT), we could just make sure we run the tests independently, would it make sense to look at a more generic property, like runtime flavor when setting this property, that way we don't need to update these with a lot of different conditions when we start to run these tests on all Mono runtime platforms supporting them?

Ah, so instead conditioning `JitOptimizationSensitive` based on `RuntimeFlavor` in each csproj? 

> we could just make sure we run the tests independently

What would this look like?

Reply:
> I'll remove all the associated `JitOptimizationSensitive` exclusions and check CI results again.

Lane `Build Android x64 Release AllSubsets_Mono_RuntimeTests` results https://helix.dot.net/api/jobs/50218d6c-c030-4a62-947e-7cfb705d399b/workitems/PayloadGroup0?api-version=2019-06-17, I see `SKIPPING EXECUTION BECAUSE COMPlus_TieredCompilation has not been disabled and this test is marked JitOptimizationSensitive
Test Harness Exitcode is : 0`

Reply:
> Ah, so instead conditioning `JitOptimizationSensitive` based on `RuntimeFlavor` in each csproj?

Yes, I believe not condition them to only Android but to MonoVM would make sense, alternative is to add the condition directly into the target that emits the script:

https://github.com/dotnet/runtime/blob/87777b624b0b8ea295b0fe9cd8505ce262ed75c0/src/tests/Common/CLRTest.Execute.Batch.targets#L96

https://github.com/dotnet/runtime/blob/87777b624b0b8ea295b0fe9cd8505ce262ed75c0/src/tests/Common/CLRTest.Execute.Bash.targets#L99

So adding something like `Condition="'$(RuntimeFlavor)' != 'mono'"` would make sure we run these tests on MonoVM independent on what CI configuration we use and what env variables that ends up setting.